Sources/SwiftyRequest/RestRequest.swift

```diff
@@ -155,45 +168,50 @@ public class RestRequest {
             return nil
         }
     }
-
+    
     /// Initialize a `RestRequest` instance
     ///
     /// - Parameters:
     ///   - url: URL string to use for network request
-    public init(method: HTTPMethod = .get, url: String) {
-
+    public init(method: HTTPMethod = .get, url: String, containsSelfSignedCert: Bool? = false) {
+        
+        self.isSecure = url.contains("https")
+        self.isSelfSigned = containsSelfSignedCert ?? false
+        
         // Instantiate basic mutable request
         let urlComponents = URLComponents(string: url) ?? URLComponents(string: "")!
         let urlObject = urlComponents.url ?? URL(string: "n/a")!
         self.request = URLRequest(url: urlObject)
-
+        
         // Set inital fields
         self.url = url
+        
+        super.init()
+        
         self.method = method
         self.acceptType = "application/json"
         self.contentType = "application/json"
-
+        
         // We accept URLs with templated values which `URLComponents` does not treat as valid
         if URLComponents(string: url) == nil {
             self.urlTemplate = url
         }
     }
-
+    
     // MARK: Response methods
-
     /// Request response method that either invokes `CircuitBreaker` or executes the HTTP request
     ///
     /// - Parameter completionHandler: Callback used on completion of operation
     public func response(completionHandler: @escaping (Data?, HTTPURLResponse?, Error?) -> Void) {
         if let breaker = circuitBreaker {
-            breaker.run(commandArgs: completionHandler, fallbackArgs: "Circuit is open")
+            breaker.run(commandArgs: "test", fallbackArgs: "Circuit is open")
```

Comment:
We need the completion handler. Also, as an aside, when we need string descriptions in our code, we should try to be more descriptive.